src/Diagnostics.DataProviders/KustoHeartBeatService/KustoHeartBeatService.cs

```diff
@@ -59,10 +60,41 @@ public string GetClusterNameFromStamp(string stampName)
 
             if (!_configuration.RegionSpecificClusterNameCollection.TryGetValue(appserviceRegion.ToLower(), out kustoClusterName))
             {
+                if(_configuration.CloudDomain == DataProviderConstants.AzureCloud)
+                {
+                    try
+                    {
+                         DataTable regionMappings = _kustoDataProvider.ExecuteClusterQuery($"cluster('wawseusfollower').database('wawsprod').WawsAn_regionsincluster | where pdate >= ago(5d) | summarize by Region, ClusterName", "RegionMappingInit").Result;
```

Comment:
`GetClusterNameFromStamp` gets called in the request pipeline by `KustoDataProvider` in the `ExecuteQuery` method. Using the `Task.Result` could block the thread. Maybe make this method an asynchronous task.

Reply:
made the change